src/main/kotlin/com/sourcegraph/cody/inspections/CodyFixHighlightPass.kt

```diff
@@ -106,15 +116,18 @@ class CodyFixHighlightPass(val file: PsiFile, val editor: Editor) :
           if (progress.isCanceled) {
             break
           }
+          if (highlight.startOffset > document.textLength ||
```

Comment:
We are now reading document in outside of the EDT.
Your check probably highly decreases chance of the issue, but I doubt it eliminates it as state can change even between this and the next instruction.
I think `document.codyRange` should be called on EDT.

Reply:
fixed